### PR TITLE
update prod workflow input templates with required booleans and workflow versions 

### PR DESCRIPTION
**Pull request by alexiswl**
## TODO
- [x] #257 
- [x] #260

## Dragen alignment with qc pipeline
* added _enable_duplicate_marking_, _enable_map_align_output_ and _enable_sort_ to qc pipeline whilst removing _sites_somalier_ and _reference_fasta_ as input keys.
* Upgrade prod version from 3.9.3--7ba55e9 to 3.9.3--4e00721
  * Removed somalier steps
  * Made _enable_duplicate_marking_ / _enable_map_align_output_ and _enable_sort_ input booleans as mandatory

## Dragen somatic pipeline
* No changes to inputs required 
* Upgrade prod version from 3.9.3--7ba55e9 to 3.9.3--4e00721
	* Made input booleans _enable_map_align_output_ and _enable_duplicate_marking_ as mandatory boolean inputs 

## Dragen transcriptome pipeline
* added _enable_map_align_output_ and _enable_duplicate_marking_ (default false) as input keys 
* Upgrade prod version from 3.9.3--7ba55e9 to 3.9.3--4e00721
	* Made input booleans _enable_map_align_output_ and _enable_duplicate_marking_ as mandatory boolean inputs 
* updated transcriptome reference directories to include quantification coverage (013cd72)

## Related PRs
* umccr/cwl-ica#151
* umccr/cwl-ica#152

## Related Issues
* https://github.com/umccr/cwl-ica/issues/120

### CODE REVIEW

**Comment by alexiswl** [comment]
@skanwal feel free to add to this commit if you wish to update the reference samples for the wts workflow prod inputs, see https://github.com/umccr/cwl-ica/issues/120 for more info.

**Comment by skanwal** [comment]
> feel free to add to this commit

I didn't understand which commit are you referring to @alexiswl ?

**Comment by alexiswl** [comment]
Sorry, feel free to add to this **branch** (or add commits to the existing commits on this branch), with the required changes for the wts reference sample directories. But you can also just put them in a separate branch.

**Comment by skanwal** [comment]
Got it. Lets keep the workflow changes under this branch. I'll push to it shortly.

**Comment by reisingerf** [comment]
@alexiswl as it's a merge into main I assume it's meant for the next portal release? 
(which would fall under a NATA/production release?)

If so, can we:

- add it to the release milestone (@victorskl do we have a cut-off date for the release yet? I thought it was already ongoing, but perhaps we can squeeze this in still?)
- add Oliver as reviewer since as a portal production release it needs to be signed off (@ohofmann this is where we define the workflow versions AND configuration for the production release)

Perhaps for the future: we could experiment with having "NATA relevant" releases (and accompanying milestones) similar to the current portal release on the workflows repo for NATA documentation, etc... and pull the releases across the GH repos together and combine them with relevant documentation? Just a thought, happy to discuss/explore...

**Comment by skanwal** [comment]
Updated in WTS workflow inputs in https://github.com/umccr/infrastructure/pull/261/commits/013cd7287f77f4d3e85cd35ba07f3c9aa5554406.
Note: I also need to update RNAsum workflow deets. It'll be helpful to keep this around until I have made changes for that workflow as well.

**Comment by alexiswl** [comment]
> add it to the release milestone (@victorskl do we have a cut-off date for the release yet? I thought it was already ongoing, but perhaps we can squeeze this in still?)

Added to Chausie, @victorskl feel free to remove if too late

> add Oliver as reviewer since as a portal production release it needs to be signed off (@ohofmann this is where we define the workflow versions AND configuration for the production release)

Done

> Perhaps for the future: we could experiment with having "NATA relevant" releases (and accompanying milestones) similar to the current portal release on the workflows repo for NATA documentation, etc... and pull the releases across the GH repos together and combine them with relevant documentation? Just a thought, happy to discuss/explore...

Yep maybe one for the devops call

**Comment by alexiswl** [comment]
> Note: I also need to update RNAsum workflow deets. It'll be helpful to keep this around until I have made changes for that workflow as well.

@skanwal when would you expect to have this completed by? I assume this is a new version of rnasum?

**Comment by skanwal** [comment]
@alexiswl - yes, I just need to generate an RNAsum tag, test in dev and update workflows with the new tag. I expect to complete this by Thursday.

**Review by victorskl — COMMENTED** [review]
> re: "NATA relevant" releases [...]

Thanks for raising it, Flo. Those are good points. We are evolving into that direction, yes. Baby step. :)

> do we have a cut-off date for the release yet? I thought it was already ongoing, but perhaps we can squeeze this in still?

Let us push Chausie release date a week more; I re-target it to 10/Nov next week Thursday. Meanwhile, I will also need a bit of time setting up Staging as well. So, if all tool and application changes are finalised by this week then we can start end-to-end Pipeline testing next week around 8/Nov Tuesday. Happy to discuss more in usual Thurs lab meeting, if any.

> Added to Chausie, victorskl feel free to remove if too late

All good, Alexis. Let us do this with Chausie, yes. 

> test in dev and update workflows with the new tag. I expect to complete this by Thursday.

Thanks for ETA, Sehrish. That would be great! :)

**Comment by victorskl** [comment]
Dear @alexiswl @skanwal -- Whenever this PR is ready/final, please kindly lift `hold merging` flag and, please mark it `Ready for review`. Thank you!

**Comment by skanwal** [comment]
Thanks @victorskl .

@alexiswl  - I have added new version of RNAsum to prod https://github.com/umccr/cwl-ica/pull/169
I'll update this PR with the new version as soon as that is merged.

That'd be the final update (hopefully) from my side.

**Comment by alexiswl** [comment]
> @alexiswl - I have added new version of RNAsum to prod https://github.com/umccr/cwl-ica/pull/169
I'll update this PR with the new version as soon as that is merged.

Approved! Should be ready to go in 30 minutes or so

**Comment by skanwal** [comment]
All done from my side.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 1]
```diff
@@ -434,7 +423,7 @@ locals {
 
   rnasum_wfl_version = {
     dev = "0.4.4"
-    prod = "0.4.4--c3e3093"
+    prod = "0.4.5--c801102"
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:426]: Noting: confirmed this version tag exists for `rnasum` workflow in PROD.

File: terraform/stacks/umccr_data_portal/workflow/main.tf [hunk 2]
```diff
@@ -245,37 +240,35 @@ locals {
...
             },
             {
                 "class": "Directory",
-                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-08-17--3.9.3/SBJ01286"
+                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01286"
             },
             {
                 "class": "Directory",
-                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-08-17--3.9.3/SBJ01673"
+                "location": "gds://development/reference-data/dragen_wts/multiqc/2022-09-14--3.9.3/SBJ01673"
```
Comment (victorskl) [inline terraform/stacks/umccr_data_portal/workflow/main.tf:259]: This update might be from #260 for DEV.

**Review by victorskl — APPROVED** [review]
LGTM! Comments are for notes/comm purpose. No actions required. 

Might want to solve conflict.

**Review by reisingerf — APPROVED** [review]
LGTM

**Comment by alexiswl** [comment]
> Might want to solve conflict.

Resolving conflict now

**Review by skanwal — APPROVED** [review]

**Comment by alexiswl** [comment]
Done

**Review by ohofmann — APPROVED** [review]
All good!

**Comment by victorskl** [comment]
Thanks all. Merging...

**Pull request merged**